oracledb/index.d.ts

```diff
@@ -1,589 +1,684 @@
-     */
-    queueRequests: number;
-    /**
-     * The time (in milliseconds) that a connection request should wait in the queue before the request is terminated.
-     */
-    queueTimeout: number;
-    /**
-     * The number of statements to be cached in the statement cache of each connection.
-     */
-    stmtCacheSize: number;
-
-    /**
-     * Finalizes the connection pool.
-     * @param  {(err:any)=>void} callback Callback called when the pool is terminated or when some error occurs
-     * @returns void
-     */
-    close(callback: (err: any) => void): void;
-
-    /**
-     * Finalizes the connection pool.
-     * @returns Promise to when the close finishes.
-     */
-    close(): IPromise<void>;
-
-    /**
-     * This method obtains a connection from the connection pool.
-     * If a previously opened connection is available in the pool, that connection is returned. If all connections in the pool are in use, a new connection is created and returned to the caller, as long as the number of connections does not exceed the specified maximum for the pool. If the pool is at its maximum limit, the getConnection() call results in an error, such as ORA-24418: Cannot open further sessions.
-     * @param  {(err:any,connection:IConnection)=>void} callback Callback called when the connection is available or when some error occurs.
-     * @returns void
-     * @see {@link https://jsao.io/2015/03/making-a-wrapper-module-for-the-node-js-driver-for-oracle-database/}
-     * @see {@link https://github.com/OraOpenSource/orawrap}
-     */
-    getConnection(callback: (err: any, connection: IConnection) => void): void;
-
-    /**
-     * This method obtains a connection from the connection pool.
-     * If a previously opened connection is available in the pool, that connection is returned. If all connections in the pool are in use, a new connection is created and returned to the caller, as long as the number of connections does not exceed the specified maximum for the pool. If the pool is at its maximum limit, the getConnection() call results in an error, such as ORA-24418: Cannot open further sessions.
-     * @returns  An IConnection Promise to when the connection is available or when some error occurs.
-     * @see {@link https://jsao.io/2015/03/making-a-wrapper-module-for-the-node-js-driver-for-oracle-database/}
-     * @see {@link https://github.com/OraOpenSource/orawrap}
-     */
-    getConnection(): IPromise<IConnection>;
-
-    /**
-     * An alias for IConnectionPool.close().
-     * @param  {(err:any)=>void} callback Callback called when the pool is terminated or when some error occurs
-     * @returns void
-     */
-    terminate(callback: (err: any) => void): void;
-
-    /**
-     * An alias for IConnectionPool.close().
-     * @returns Promise to when the close finishes.
-     */
-    terminate(): IPromise<void>;
+import * as stream from 'stream';
+import * as events from 'events';
+
+declare namespace oracledb {
```

Comment:
And the repo I created to update this definitions is [here](https://github.com/Bigous/typed-oracledb)